src/core/Banner/style.ts

```diff
@@ -33,14 +40,17 @@ export const IconWrapper = styled("div")`
   }}
 `;
 
-type ButtonIconType = ButtonIconExtraProps & { bannerType: string };
+type ButtonIconType<ButtonIconSize extends keyof ButtonIconSizeToTypes> =
```

Comment:
I feel like this type is being used p often, wondering if it should be a named type